doc/ansible/user-guide.md

```diff
@@ -0,0 +1,279 @@
+- As go program outside cluster using `operator-sdk`
+
+#### 1. Run as pod inside a Kubernetes cluster
+
+Run as pod inside a Kubernetes cluster is preferred for production use.
+
+Build the memcached-operator image and push it to a registry:
+```
+$ operator-sdk build quay.io/example/memcached-operator:v0.0.1
+$ sed -i 's|REPLACE_IMAGE|quay.io/example/memcached-operator:v0.0.1|g' deploy/operator.yaml
+$ docker push quay.io/example/memcached-operator:v0.0.1
+```
+
+Kubernetes deployment manifests are generated in `deploy/operator.yaml`. The deployment image is set to the container image specified above.
+
+Deploy the memcached-operator:
+
+```sh
+$ kubectl create -f deploy/rbac.yaml
+$ kubectl create -f deploy/operator.yaml
+```
+
+Verify that the memcached-operator is up and running:
+
+```sh
+$ kubectl get deployment
+NAME                     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+memcached-operator       1         1         1            1           1m
+```
+
+#### 2. Run outside the cluster
+
+This method is preferred during development cycle to deploy and test faster.
+
+Run the operator locally with the default kubernetes config file present at `$HOME/.kube/config`:
+
+```sh
+$ operator-sdk up local
+INFO[0000] Go Version: go1.10
+INFO[0000] Go OS/Arch: darwin/amd64
+INFO[0000] operator-sdk Version: 0.0.5+git
+```
+
+Run the operator locally with a provided kubernetes config file:
+
+```sh
+$ operator-sdk up local --kubeconfig=config
+INFO[0000] Go Version: go1.10
+INFO[0000] Go OS/Arch: darwin/amd64
+INFO[0000] operator-sdk Version: 0.0.5+git
+```
+
+### Create a Memcached CR
+
+Modify `deploy/cr.yaml` as shown and create a `Memcached` custom resource:
+
+```sh
+$ cat deploy/cr.yaml
+apiVersion: "cache.example.com/v1alpha1"
+kind: "Memcached"
+metadata:
+  name: "example-memcached"
+spec:
+  size: 3
+
+$ kubectl apply -f deploy/cr.yaml
+```
+
+Ensure that the memcached-operator creates the deployment for the CR:
+
+```sh
+$ kubectl get deployment
+NAME                     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+memcached-operator       1         1         1            1           2m
+example-memcached        3         3         3            3           1m
+```
+
+Check the pods to confirm 3 replicas were created:
+
+```sh
+$ kubectl get pods
+NAME                                  READY     STATUS    RESTARTS   AGE
+example-memcached-6fd7c98d8-7dqdr     1/1       Running   0          1m
+example-memcached-6fd7c98d8-g5k7v     1/1       Running   0          1m
+example-memcached-6fd7c98d8-m7vn7     1/1       Running   0          1m
+memcached-operator-7cc7cfdf86-vvjqk   1/1       Running   0          2m
+```
+
+### Update the size
+
+Change the `spec.size` field in the memcached CR from 3 to 4 and apply the change:
+
+```sh
+$ cat deploy/cr.yaml
+apiVersion: "cache.example.com/v1alpha1"
+kind: "Memcached"
+metadata:
+  name: "example-memcached"
+spec:
+  size: 4
+
+$ kubectl apply -f deploy/cr.yaml
```

Comment:
You could just use the `kubectl patch` command here to make things simpler I think?

Reply:
In the user-guide that is at the top-level they seem to suggest using `apply`. Any idea what the difference between them are?